### PR TITLE
fix no data type

### PR DESCRIPTION
**Pull request by michaelmattig**
https://trello.com/c/k6Q3FS3T

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Operator: Daten in angegebene Datentyp aus Parametern konvertieren (aktuell: Fehler)](https://trello.com/c/k6Q3FS3T/275-operator-daten-in-angegebene-datentyp-aus-parametern-konvertieren-aktuell-fehler)

File: projects/wave-core-new/src/lib/operators/datatype.model.ts [hunk 1]
```diff
@@ -22,6 +24,8 @@ export abstract class DataType {
      * @return the smallest value.
      */
     abstract getMax(): number;
+
+    abstract no_data(value: number): NoDataDict;
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/operators/datatype.model.ts:28]: I also struggle with this, but I guess we should use camel case in our JS world :grinning: .

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]
You need to merge the master.

Furthermore, do you get any reasonable output for NDVI with data that is not Byte?
For instance, when you set the data type to Unsigned Int 16 and calculate A * 256, shouldn't you get a similar output to the original?

File: projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts [hunk 2]
```diff
@@ -211,9 +211,7 @@ export class ExpressionOperatorComponent implements AfterViewInit, OnDestroy {
                     params: {
                         expression,
                         output_type: dataType.getCode(),
-                        output_no_data_value: {
-                            U8: 42 // TODO: make this configurable once units exist again
-                        }
+                        output_no_data_value: dataType.no_data(42),  // TODO: make this configurable once units exist again
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts:214]: maybe use datatype MAX or MAX for NODATA in the meantime?

File: projects/wave-core-new/src/lib/operators/datatype.model.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/operators/datatype.model.ts]: 0365c2d4a4bb46141542e9a67e1c4f961c2e097a

File: projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts (hunk 2 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts]: 0365c2d4a4bb46141542e9a67e1c4f961c2e097a

**Comment by michaelmattig** [comment]
> 
> 
> You need to merge the master.
> 
> Furthermore, do you get any reasonable output for NDVI with data that is not Byte?
> For instance, when you set the data type to Unsigned Int 16 and calculate A * 2, shouldn't you get a similar output to the original?

The bigger the data type, the more grey/black the output image gets. I figured it's a problem with the colorizer who uses the data type range but maybe it's something else.

**Pull request merged**